File: itx-java/itx-impl/src/main/java/io/github/chehsunliu/itx/impl/Env.java
package io.github.chehsunliu.itx.impl;

public final class Env {
  private Env() {}

  public static String requireEnv(String name) {
    String v = System.getenv(name);
    if (v == null) throw new IllegalStateException("missing env var: " + name);
    return v;
  }

  public static int envInt(String name, int defaultValue) {
    String v = System.getenv(name);
    if (v == null || v.isBlank()) return defaultValue;
    return Integer.parseInt(v);
  }
}

File: itx-kt/itx-impl/src/main/kotlin/io/github/chehsunliu/itx/impl/Env.kt
package io.github.chehsunliu.itx.impl

internal fun requireEnv(name: String): String = System.getenv(name) ?: error("missing env var: $name")

internal fun envInt(
    name: String,
    default: Int,
): Int = System.getenv(name)?.takeIf { it.isNotBlank() }?.toInt() ?: default
